Wrong value for project.root.dir property

### Describe the bug

`project.root.dir` property, defined in parent pom, relies on `maven.multiModuleProjectDirectory`, which is not documented and might fail as explained [here](https://stackoverflow.com/questions/29778262/what-is-maven-multimoduleprojectdirectory-used-for).
It should be replaced by proper usage of [ Maven directory plugin](https://github.com/jdcasey/directory-maven-plugin?tab=readme-ov-file#highest-basedir-goal)


### Expected behavior

_No response_

### Actual behavior

In any enviroment where test are not invoked from kogito-runtimes root, this causes test errors in modules using this property,
For example
```
[ERROR] Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.061 s <<< FAILURE! -- in org.kie.kogito.quarkus.workflows.CallbackStateIT
[ERROR] org.kie.kogito.quarkus.workflows.CallbackStateIT.callbackStateWithError -- Time elapsed: 0.039 s <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Expected status code <201> but was <404>.
``` 
In working scenarios, The callback json file is copied into the Integration test module from the Workflow builder module. When `project.root.dir `is wrong, the file is not found while copying,  hence is not there when the test is run, causing the not found failure. 



### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Kogito version or git rev (or at least Quarkus version if you are using Kogito via Quarkus platform BOM)

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/kogito-build/kogito-build-no-bom-parent/pom.xml
+++ b/kogito-build/kogito-build-no-bom-parent/pom.xml
@@ -673,7 +673,7 @@
         <plugin>
           <groupId>org.commonjava.maven.plugins</groupId>
           <artifactId>directory-maven-plugin</artifactId>
-          <version>0.1</version>
+          <version>1.0</version>
           <executions>
             <execution>
               <id>directories</id>

--- a/kogito-build/kogito-build-no-bom-parent/pom.xml
+++ b/kogito-build/kogito-build-no-bom-parent/pom.xml
@@ -87,7 +87,6 @@
       possible to measure cross-module test coverage -->
     <jacoco.exec.file>${project.root.dir}/target/jacoco.exec</jacoco.exec.file>
     <!--suppress UnresolvedMavenProperty -->
-    <project.root.dir>${maven.multiModuleProjectDirectory}</project.root.dir>
     <quarkus.logging.manager>org.jboss.logmanager.LogManager</quarkus.logging.manager>
     <tests.surefire.include>**/*Test.java</tests.surefire.include>
     <tests.failsafe.include>**/*IT.java</tests.failsafe.include>
@@ -671,6 +670,23 @@
             <legacyMode>true</legacyMode>
           </configuration>
         </plugin>
+        <plugin>
+          <groupId>org.commonjava.maven.plugins</groupId>
+          <artifactId>directory-maven-plugin</artifactId>
+          <version>0.1</version>
+          <executions>
+            <execution>
+              <id>directories</id>
+              <goals>
+                <goal>highest-basedir</goal>
+               </goals>
+              <phase>initialize</phase>
+              <configuration>
+                <property>project.root.dir</property>
+               </configuration>
+             </execution>
+          </executions>
+        </plugin>
       </plugins>
     </pluginManagement>
 
@@ -705,6 +721,10 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-checkstyle-plugin</artifactId>
       </plugin>
+      <plugin>
+        <groupId>org.commonjava.maven.plugins</groupId>
+        <artifactId>directory-maven-plugin</artifactId>
+       </plugin>
     </plugins>
   </build>
 

--- a/kogito-build/kogito-build-no-bom-parent/pom.xml
+++ b/kogito-build/kogito-build-no-bom-parent/pom.xml
@@ -680,7 +680,7 @@
               <goals>
                 <goal>highest-basedir</goal>
                </goals>
-              <phase>initialize</phase>
+              <phase>validate</phase>
               <configuration>
                 <property>project.root.dir</property>
                </configuration>
@@ -691,6 +691,10 @@
     </pluginManagement>
 
     <plugins>
+       <plugin>
+        <groupId>org.commonjava.maven.plugins</groupId>
+        <artifactId>directory-maven-plugin</artifactId>
+       </plugin>
       <plugin>
         <groupId>net.revelc.code.formatter</groupId>
         <artifactId>formatter-maven-plugin</artifactId>
@@ -721,10 +725,6 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-checkstyle-plugin</artifactId>
       </plugin>
-      <plugin>
-        <groupId>org.commonjava.maven.plugins</groupId>
-        <artifactId>directory-maven-plugin</artifactId>
-       </plugin>
     </plugins>
   </build>
 

--- a/quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-image-integration-test/pom.xml
+++ b/quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-image-integration-test/pom.xml
@@ -87,7 +87,7 @@
         <directory>src/main/resources</directory>
       </resource>
       <resource>
-        <directory>../kogito-quarkus-serverless-workflow-integration-test/src/main/resources</directory>
+        <directory>${project.root.dir}/quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-integration-test/src/main/resources</directory>
         <includes>
           <include>expression.sw.json</include>
           <include>schema/expressionInput.json</include>

--- a/quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-image-integration-test/pom.xml
+++ b/quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-image-integration-test/pom.xml
@@ -87,7 +87,7 @@
         <directory>src/main/resources</directory>
       </resource>
       <resource>
-        <directory>${project.root.dir}/quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-integration-test/src/main/resources</directory>
+        <directory>../kogito-quarkus-serverless-workflow-integration-test/src/main/resources</directory>
         <includes>
           <include>expression.sw.json</include>
           <include>schema/expressionInput.json</include>

